app/views/spree/checkout/payment/_affirm.html.erb

```diff
@@ -7,4 +7,5 @@
   payload = Solidus::Affirm::CheckoutPayload.new(current_order, config)
   json = Solidus::Affirm::CheckoutPayloadSerializer.new(payload, root: false).to_json
 %>
-<div id="affirm_checkout_payload" class="hidden" data-affirm='<%=raw(json)%>' />
+
+<div id="affirm_checkout_payload" class="hidden" data-affirm='<%=raw(json)%>' data-paymentgateway='<%=Spree::Gateway::Affirm.first.id%>' />
```

Comment:
While it isn't happening for spree frontend, because spree frontend isn't dynamic, order information can change while you're on the checkout page. It's better to have an API endpoint that returns the affirm payload as its needed, so you're guaranteed to have the most accurate data at the time.